src/cocoa/toga_cocoa/widgets/tree.py

```diff
@@ -210,11 +219,19 @@ def create(self):
     #         else:
     #             self.tree.collapseItem(node._impl)
 
+    # TODO: don't reload the entire view if not necessary
+    #  reloadData is slow and invalidates any current selections
+
     def change_source(self, source):
         self.tree.reloadData()
 
     def insert(self, parent, index, item):
-        self.tree.reloadData()
+        index_set = NSIndexSet.indexSetWithIndex(index)
+        if parent == self.interface.data:
+            self.tree.reloadData()
+        else:
+            parent = id(parent._impl)
+            self.tree.insertItemsAtIndexes_inParent_withAnimation_(index_set, parent, NSTableViewAnimation.SlideDown.value)
```

Comment:
Ah, that is good to know. I have been very annoyed about all the underscores :)

Reply:
There was actually a nasty bug in this code, which I didn't see because the 'file explorer' example only inserts items before a row is expanded: The ``parent`` argument should not be ``id(parent._impl)`` but just ``parent._impl``.

The regular 'tree' example tests this very well.